frame/vamm/src/lib.rs

```diff
@@ -313,32 +308,30 @@ pub mod pallet {
 
 		#[transactional]
 		fn create(
-			base_asset_reserves: Self::Balance,
-			quote_asset_reserves: Self::Balance,
-			peg_multiplier: Self::Balance,
+			params: composable_traits::vamm::VammParams<BalanceOf<T>>,
 		) -> Result<Self::VammId, DispatchError> {
 			// TODO: (Matheus)
 			// How to ensure that the caller has the right privileges?
 			// (eg. How to ensure the caller is the Clearing House, and not anyone else?)
-			ensure!(!base_asset_reserves.is_zero(), Error::<T>::BaseAssetReserveIsZero);
-			ensure!(!quote_asset_reserves.is_zero(), Error::<T>::QuoteAssetReserveIsZero);
-			ensure!(!peg_multiplier.is_zero(), Error::<T>::PegMultiplierIsZero);
+			ensure!(!params.base_asset_reserves.is_zero(), Error::<T>::BaseAssetReserveIsZero);
+			ensure!(!params.quote_asset_reserves.is_zero(), Error::<T>::QuoteAssetReserveIsZero);
+			ensure!(!params.peg_multiplier.is_zero(), Error::<T>::PegMultiplierIsZero);
 
-			VammCounter::<T>::try_mutate(|id| {
-				let old_id = *id;
+			VammCounter::<T>::try_mutate(|next_id| {
+				let id = *next_id;
```

Comment:
Hmm, cool. No need to call `.clone()`!

Reply:
Ah, it's because it has the `Copy` trait

Reply:
Yep haha